server/routes/activities/allocation-dashboard/handlers/allocationDashboard.ts

```diff
@@ -170,6 +172,42 @@ export default class AllocationDashboardRoutes {
     })
   }
 
+  private getWaitlistedPrisoners = async (scheduleId: number, user: ServiceUser) => {
+    const waitlist = await this.activitiesService
+      .fetchActivityWaitlist(scheduleId, user)
+      .then(a => a.filter(w => w.status === 'PENDING' || w.status === 'APPROVED' || w.status === 'DECLINED'))
+      .then(a => a.filter(w => !w.allocationId))
+
+    const prisonerNumbers = waitlist.map(application => application.prisonerNumber)
+    const [inmateDetails, prisonerAllocations]: [Prisoner[], PrisonerAllocations[]] = await Promise.all([
+      this.prisonService.searchInmatesByPrisonerNumbers(prisonerNumbers, user),
+      this.activitiesService.getActivePrisonPrisonerAllocations(prisonerNumbers, user),
+    ])
+
+    return inmateDetails.flatMap(inmate => {
+      const thisWaitlist = waitlist.filter(a => a.prisonerNumber === inmate.prisonerNumber)
+      let otherAllocations: Allocation[] = []
+      if (prisonerAllocations.length > 0) {
+        otherAllocations = prisonerAllocations
+          .find(a => a.prisonerNumber === inmate.prisonerNumber)
+          ?.allocations.filter(a => a.scheduleId !== scheduleId)
+      }
+      return thisWaitlist.map(w => ({
+        waitlistApplicationId: w.id,
+        name: `${inmate.firstName} ${inmate.lastName}`,
+        prisonerNumber: inmate.prisonerNumber,
+        cellLocation: inmate.cellLocation,
+        requestDate: parseDate(w.requestedDate),
+        requestedBy: w.requestedBy,
+        status: w.status,
+        otherAllocations: otherAllocations?.map(a => ({
+          id: a.scheduleId,
+          scheduleName: a.scheduleDescription,
+        })),
+      }))
+    })
+  }
```

Comment:
I can see there's no pagination on the waitlist people, but there might be issues with pagination anyway, as we'd be filtering people out of the pagination source, so looks fine.